test/src/test/java/hudson/model/UpdateCenterErrorsTest.java

```diff
@@ -0,0 +1,185 @@
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.assertEquals;
+
+public class UpdateCenterErrorsTest {
+    private static final Logger LOGGER = Logger.getLogger(Jenkins.class.getName());
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    /**
+     * Check if the page contains the right message after checking an update site with an url that returns a 502 error code.
+     * @throws Exception If there are errors managing the web elements.
+     */
+    @Test
+    public void updateSiteReturn502Test() throws Exception {
+        checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSite502/getJson", "Server returned HTTP response code: 502 for URL", false);
+    }
+
+    /**
+     * Check if the page contains the right message after checking an update site with an url that returns a wrong json.
+     * @throws Exception If there are errors managing the web elements.
+     */
+    @Test
+    public void updateSiteWrongJsonTest() throws Exception {
+        checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSiteWrongJson/getJson", "Unquotted string 'wrongjson'", false);
+    }
+
+    /**
+     * Check if the page contains the right message after checking an update site that returns a well defined json.
+     * @throws Exception If there are errors managing the web elements.
+     */
+    @Test
+    public void updateSiteRightJsonTest() throws Exception {
+        //Avoid CertPathValidatorException: Algorithm constraints check failed on signature algorithm: MD5withRSA
+        DownloadService.signatureCheck = false;
+        // Have to end in update-center.json or it fails. See UpdateSite#getMetadataUrlForDownloadable
+        checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSiteRightJson/update-center.json", "", true );
+    }
+
+    private HtmlButton getCheckNow(HtmlPage page){
+        DomNodeList<HtmlElement> elements = page.getElementById("bottom-sticker").getElementsByTagName("button");
+        assertEquals(1, elements.size());
+        return (HtmlButton) elements.get(0);
+    }
+
+    /**
+     * Check the update site.
+     * @param urlUpdateSite If null, use the default update site, otherwise, use this update site.
+     * @param message The message that should exist or not in the page after checking the update site.
+     * @param isSuccess If true, test that PluginManager.CheckUpdateSesrverError + message doesn't exist in the page.
+     *                  If false, test that PluginManager.CheckUpdateSesrverError + message exists in the page.
+     * @throws IOException If an exception is thrown using the UI.
+     * @throws SAXException If an exception is thrown using the UI.
+     */
+    private void checkUpdateSite(String urlUpdateSite, String message, boolean isSuccess) throws IOException, SAXException {
+        if(urlUpdateSite != null) {
```

Comment:
nit:
```suggestion
        if (urlUpdateSite != null) {
```

Reply:
Actually I think it can be removed, since it's never null?